ciphersuite/ed25519.go

```diff
@@ -0,0 +1,221 @@
+package ciphersuite
+
+import (
+	"encoding/hex"
+	"io"
+
+	"golang.org/x/crypto/ed25519"
+	"golang.org/x/xerrors"
+)
+
+const errNotEd25519CipherSuite = "invalid cipher suite"
+const errInvalidBufferSize = "invalid buffer size"
+
+// Ed25519CipherSuiteName is the name of the cipher suite that is using Ed25519 as
+// the signature algorithm.
+const Ed25519CipherSuiteName = "ED25519_CIPHER_SUITE"
+
+// Ed25519PublicKey is the public key implementation for the Ed25519 cipher suite.
+type Ed25519PublicKey struct {
+	data ed25519.PublicKey
+}
+
+// Name returns the name of the cipher suite.
+func (pk *Ed25519PublicKey) Name() Name {
+	return Ed25519CipherSuiteName
+}
+
+func (pk *Ed25519PublicKey) String() string {
+	return hex.EncodeToString(pk.data)
```

Comment:
Why stringify (and not the same format as the RawPublicKey)?

Reply:
And some others in the same file.

Reply:
Good point